src/components/CouponDetails/ActionButton.jsx

```diff
@@ -0,0 +1,124 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {
+  Button,
+} from '@edx/paragon';
+import RemindButton from '../RemindButton';
+import RevokeButton from '../RevokeButton';
+import { ACTION_TYPES, ACTION_LABELS, COUPON_FILTER_TYPES } from './constants';
+
+const ActionButton = ({
+  couponData: {
+    id,
+    errors,
+    available,
+    title,
+  },
+  selectedToggle,
+  code,
+  handleCodeActionSuccess,
+  setModalState,
+}) => {
+  const {
+    assigned_to: assignedTo,
+    is_public: isPublic,
+    redemptions,
+  } = code;
+  let remainingUses = redemptions.total - redemptions.used;
+
+  // Don't show `Assign/Remind/Revoke` buttons for an unavailable coupon
+  if (!available) {
+    return null;
+  }
+
+  // Don't show a button if all total redemptions have been used
+  if (redemptions.used === redemptions.total) {
+    return null;
+  }
+
+  const codeHasError = errors.find(errorItem => errorItem.code === code.code);
+  if (assignedTo) {
+    return (
+      <>
+        {!codeHasError && (
+          <>
+            <RemindButton
+              couponId={id}
+              couponTitle={title}
+              data={{
+                code: code.code,
+                email: code.assigned_to,
+              }}
+              onSuccess={response => handleCodeActionSuccess(ACTION_TYPES.remind, response)}
+            />
+            {' | '}
+          </>
+        )}
+        <RevokeButton
+          couponId={id}
+          couponTitle={title}
+          data={{
+            assigned_to: code.assigned_to,
+            code: code.code,
+          }}
+          onSuccess={response => handleCodeActionSuccess(ACTION_TYPES.revoke, response)}
+        />
+      </>
+    );
+  }
+
+  if (isPublic) {
```

Comment:
Nit: can we put the null logic for 'isPublic' up at the top with all the other null logics?

Reply:
Possibly. I feel like there might have been a reason for this here, but I will double check it.